src/main/java/org/spongepowered/api/adventure/Audiences.java

```diff
@@ -47,7 +47,18 @@ public static Audience server() {
      * @return An audience
      */
     public static Audience onlinePlayers() {
-        return factory().onlinePlayers();
+        return Sponge.getRegistry().getFactoryRegistry().provideFactory(Factory.class).onlinePlayers();
+    }
+
+    /**
+     * Gets an {@link Audience} that targets all online players
+     * with the specified permission.
+     *
+     * @param permission The permission
+     * @return An audience
+     */
+    public static Audience withPermission(String permission) {
```

Comment:
`final` and `Objects.requireNonNull` in the method to fail out as soon as possible if there is a problem.